packages/vanilla-renderers/src/complex/categorization/CategorizationList.tsx

```diff
@@ -24,55 +24,65 @@
 */
 import React, { useMemo } from 'react';
 import {
-  Categorization,
   Category,
+  Categorization,
   deriveLabelForUISchemaElement,
   Translator,
+  isVisible,
 } from '@jsonforms/core';
 import { isCategorization } from './tester';
+import { AjvProps } from '../../util';
 
 const getCategoryClassName = (
   category: Category,
   selectedCategory: Category
 ): string => (selectedCategory === category ? 'selected' : '');
 
 export interface CategorizationProps {
-  categorization: Categorization;
+  elements: (Category | Categorization)[];
   selectedCategory: Category;
   depth: number;
+  data: any;
   onSelect: any;
   subcategoriesClassName: string;
   groupClassName: string;
   t: Translator;
 }
 
 export const CategorizationList = ({
-  categorization,
   selectedCategory,
+  elements,
+  data,
   depth,
   onSelect,
   subcategoriesClassName,
   groupClassName,
   t,
-}: CategorizationProps) => {
+  ajv,
+}: CategorizationProps & AjvProps) => {
+  const filteredElements = elements.filter(
```

Comment:
This should be memoed via a `useMemo`. Otherwise, the memo for the `categoryLabels` is useless because `filteredElements` is recalculated on each render and, thus, breaks the memo of `categoryLabels`.